accountsplus/settings.py

```diff
@@ -0,0 +1,70 @@
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.apps import apps
+
+
+def get_setting(setting_str, default_value=None):
+    try:
+        return getattr(settings, setting_str, default_value)
+    except Exception:
+        raise ImproperlyConfigured('Setting {0} is not configured'.format(setting_str))
+
+
+def get_captcha_public_key():
+    return get_setting('RECAPTCHA_PUBLIC_KEY')
+
+
+def get_captcha_private_key():
+    return get_setting('RECAPTCHA_PRIVATE_KEY')
+
+
+def is_ssl_enabled_for_captcha():
+    return get_setting('RECAPTCHA_USE_SSL')
+
+
+def get_cooloff_time():
+    return get_setting('AXES_COOLOFF_TIME')
+
+
+def get_login_failure_limit():
+    return get_setting('AXES_LOGIN_FAILURE_LIMIT')
+
+
+def get_lockout_url():
+    return get_setting('AXES_LOCKOUT_URL')
+
+
+def get_enable_lockout():
+    return get_setting('AXES_LOCK_OUT_AT_FAILURE')
+
+
+LOCKOUT_TEMPLATE = 'accounts/locked_out.html'
+
+def get_lockout_template():
+    return get_setting('AXES_LOCKOUT_TEMPLATE', LOCKOUT_TEMPLATE)
+
+
+ENABLE_LOCKOUT = bool(get_enable_lockout())
+
+if ENABLE_LOCKOUT:
+    # Check if the required apps are installed
+    if not apps.is_installed('axes'):
+        raise ImproperlyConfigured('axes is not configured in your INSTALLED_APPS')
+    if not apps.is_installed('captcha'):
+        raise ImproperlyConfigured('captcha is not configured in your INSTALLED_APPS')
+
+    # Checking if those parameters are configured within the app settings
+    PRIVATE_KEY = get_captcha_private_key()
+    PUBLIC_KEY = get_captcha_private_key()
+
+    # Checking if recaptcha is enabled over SSL
+    RECAPTCHA_USE_SSL = bool(is_ssl_enabled_for_captcha())
+    if not RECAPTCHA_USE_SSL:
+        raise ImproperlyConfigured('Recaptcha is not configured over SSL')
+
+    COOLOFF_TIME = int(get_cooloff_time())
+    LOGIN_FAILURE_LIMIT = int(get_login_failure_limit())
+    LOCKOUT_URL = str(get_lockout_url())
+    if not LOCKOUT_URL or LOCKOUT_URL != 'locked/':
```

Comment:
@nasief This way we are forcing users to set LOCKOUT_URL to `locked/` which doesn't make sense at all. We should have a default value of `locked/` and at the same time we let users to change this value, and add the new value to their `urls.py`